src/diffusers/pipelines/auto_pipeline.py

```diff
@@ -0,0 +1,910 @@
+    AUTO_INPAINTING_PIPELINES_MAPPING,
+]
+
+
+def _get_task_class(mapping, pipeline_class_name):
+    def get_model(pipeline_class_name):
+        for task_mapping in SUPPORTED_TASKS_MAPPINGS:
+            for model_name, pipeline in task_mapping.items():
+                if pipeline.__name__ == pipeline_class_name:
+                    return model_name
+
+    model_name = get_model(pipeline_class_name)
+
+    if model_name is not None:
+        task_class = mapping.get(model_name, None)
+        if task_class is not None:
+            return task_class
+    raise ValueError(
+        f"AutoPipeline can't find a pipeline linked to {pipeline_class_name} for {model_name} in {mapping}"
+    )
+
+
+def _get_signature_keys(obj):
+    parameters = inspect.signature(obj.__init__).parameters
+    required_parameters = {k: v for k, v in parameters.items() if v.default == inspect._empty}
+    optional_parameters = set({k for k, v in parameters.items() if v.default != inspect._empty})
+    expected_modules = set(required_parameters.keys()) - {"self"}
+    return expected_modules, optional_parameters
+
+
+class AutoPipelineForText2Image(ConfigMixin):
+    r"""
+
+    AutoPipeline for text-to-image generation.
+
+    [`AutoPipelineForText2Image`] is a generic pipeline class that will be instantiated as one of the text-to-image
+    pipeline class in diffusers.
+
+    The pipeline type (for example [`StableDiffusionPipeline`]) is automatically selected when created with the
+    AutoPipelineForText2Image.from_pretrained(pretrained_model_name_or_path) or
+    AutoPipelineForText2Image.from_pipe(pipeline) class methods .
+
+    This class cannot be instantiated using __init__() (throws an error).
+
+    Class attributes:
+
+        - **config_name** (`str`) -- The configuration filename that stores the class and module names of all the
+          diffusion pipeline's components.
+
+    """
+    config_name = "model_index.json"
+
+    def __init__(self, *args, **kwargs):
+        raise EnvironmentError(
+            f"{self.__class__.__name__} is designed to be instantiated "
+            f"using the `{self.__class__.__name__}.from_pretrained(pretrained_model_name_or_path)` or "
+            f"`{self.__class__.__name__}.from_pipe(pipeline)` methods."
+        )
+
+    @classmethod
+    def from_pretrained(cls, pretrained_model_or_path, **kwargs):
+        r"""
+        Instantiates a text-to-image Pytorch diffusion pipeline from pretrained pipeline weight.
+
+        The from_pretrained() method takes care of returning the correct pipeline class instance by:
+            1. Detect the pipeline class of the pretrained_model_or_path based on the _class_name property of its
+               config object
+            2. Find the text-to-image pipeline linked to the pipeline class using pattern matching on pipeline class
+               name.
+
+        If a `controlnet` argument is passed, it will instantiate a StableDiffusionControlNetPipeline object.
```

Comment:
```suggestion
        If a `controlnet` argument is passed, it will instantiate a [`StableDiffusionControlNetPipeline`] object.
```

So that it's auto-hyperlinked.